timeseries/src/autogluon/timeseries/models/local/abstract_local_model.py

```diff
@@ -88,69 +100,97 @@ def _fit(self, train_data: TimeSeriesDataFrame, time_limit: int = None, **kwargs
+            data = data.groupby(level=ITEMID, sort=False).tail(self.max_ts_length)
+
+        df = pd.DataFrame(data).reset_index(level=ITEMID)
+        all_series = (ts for _, ts in df.groupby(by=ITEMID, as_index=False, sort=False)[self.target])
+
+        # timeout ensures that no individual job takes longer than time_limit
+        # TODO: a job started late may still exceed time_limit - how to prevent that?
+        timeout = None if self.n_jobs == 1 else self.time_limit
+        # end_time ensures that no new jobs are started after time_limit is exceeded
+        end_time = None if self.time_limit is None else time.time() + self.time_limit
+        executor = Parallel(self.n_jobs, timeout=timeout)
+
+        try:
+            with statsmodels_joblib_warning_filter(), statsmodels_warning_filter():
+                predictions = executor(delayed(self._predict_wrapper)(ts, end_time=end_time) for ts in all_series)
+        except TimeoutError:
+            raise TimeLimitExceeded
+
+        predictions_df = pd.concat(predictions)
+        predictions_df.index = get_forecast_horizon_index_ts_dataframe(data, self.prediction_length)
+        return TimeSeriesDataFrame(predictions_df)
 
-        if self.freq != data.freq:
-            raise RuntimeError(
-                f"{self.name} has frequency '{self.freq}', which doesn't match the frequency "
-                f"of the dataset '{data.freq}'."
+    def score_and_cache_oof(
+        self, val_data: TimeSeriesDataFrame, store_val_score: bool = False, store_predict_time: bool = False
+    ) -> None:
+        super().score_and_cache_oof(val_data, store_val_score, store_predict_time)
+        # Remove time_limit for future predictions
+        self.time_limit = None
+
+    def _predict_wrapper(self, time_series: pd.Series, end_time: Optional[float] = None) -> pd.DataFrame:
+        if end_time is not None and time.time() >= end_time:
+            raise TimeLimitExceeded
+        try:
+            result = self._predict_with_local_model(
+                time_series=time_series,
+                local_model_args=self._local_model_args.copy(),
             )
-        if self.MAX_TS_LENGTH is not None:
-            logger.debug(f"Shortening all time series to at most {self.MAX_TS_LENGTH}")
-            data = data.groupby(level=ITEMID, sort=False).tail(self.MAX_TS_LENGTH)
-
-        self._fit_and_cache_predictions(data, quantile_levels=quantile_levels)
-        item_id_to_prediction = {}
-        for item_id, ts_hash in hash_ts_dataframe_items(data).items():
-            item_id_to_prediction[item_id] = self._cached_predictions[ts_hash]
-        predictions_df = pd.concat(item_id_to_prediction)
-        predictions_df.index.rename([ITEMID, TIMESTAMP], inplace=True)
-        return TimeSeriesDataFrame(predictions_df)
+        except Exception as e:
+            logger.debug(f"{self.name} failed with exception {e}, falling back to SeasonalNaive")
```

Comment:
Let's log `error` instead of `debug`, as this suggests potential issues in the stats model implementation that should not go unnoticed.